maven-extension/src/main/java/io/opentelemetry/maven/OtelTransferListener.java

```diff
@@ -0,0 +1,161 @@
+ */
+
+package io.opentelemetry.maven;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.maven.semconv.MavenOtelSemanticAttributes;
+import io.opentelemetry.semconv.HttpAttributes;
+import io.opentelemetry.semconv.ServerAttributes;
+import io.opentelemetry.semconv.UrlAttributes;
+import io.opentelemetry.semconv.incubating.HttpIncubatingAttributes;
+import io.opentelemetry.semconv.incubating.UrlIncubatingAttributes;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.maven.execution.ExecutionListener;
+import org.apache.maven.execution.MavenSession;
+import org.eclipse.aether.transfer.AbstractTransferListener;
+import org.eclipse.aether.transfer.TransferEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Don't mark this class as {@link javax.inject.Named} and {@link javax.inject.Singleton} because
+ * Maven Sisu doesn't automatically load instance of {@link ExecutionListener} as Maven Extension
+ * hooks the same way Maven Plexus did so we manually hook this instance of {@link
+ * ExecutionListener} through the {@link OtelLifecycleParticipant#afterProjectsRead(MavenSession)}.
+ */
+public final class OtelTransferListener extends AbstractTransferListener {
+
+  private static final Logger logger = LoggerFactory.getLogger(OtelTransferListener.class);
+
+  private final SpanRegistry spanRegistry;
+
+  private final OpenTelemetrySdkService openTelemetrySdkService;
+
+  private final Map<String, Optional<URI>> repositoryUriMapping = new ConcurrentHashMap<>();
+
+  OtelTransferListener(SpanRegistry spanRegistry, OpenTelemetrySdkService openTelemetrySdkService) {
+    this.spanRegistry = spanRegistry;
+    this.openTelemetrySdkService = openTelemetrySdkService;
+  }
+
+  @Override
+  public void transferInitiated(TransferEvent event) {
+    logger.info("OpenTelemetry: OtelTransferListener#transferInitiated({})", event);
+
+    String httpRequestMethod;
+    switch (event.getRequestType()) {
+      case PUT:
+        httpRequestMethod = "PUT";
+        break;
+      case GET:
+        httpRequestMethod = "GET";
+        break;
+      case GET_EXISTENCE:
+        httpRequestMethod = "HEAD";
+        break;
+      default:
+        logger.warn("OpenTelemetry: Unknown request type {}", event.getRequestType());
+        httpRequestMethod = event.getRequestType().name();
+    }
+
+    String urlTemplate =
+        event.getResource().getRepositoryUrl()
+            + "$groupId/$artifactId/$version/$artifactId-$version.$classifier";
+
+    String spanName = httpRequestMethod + " " + urlTemplate;
+
+    SpanBuilder spanBuilder =
+        this.openTelemetrySdkService
+            .getTracer()
+            .spanBuilder(spanName)
```

Comment:
We could add a comment indicating we chose to create "http client spans" assuming the underlying http lib is not instrumented (span kind client...). 
If we instrument the http lib, then we will have to change this spn type.

Reply:
Fix proposed in:
* https://github.com/smurf667/opentelemetry-java-contrib/pull/2